src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Resources.resx

```diff
@@ -595,4 +595,7 @@
   <data name="Cannot_add_item_with_key_0_to_an_existing_1" xml:space="preserve">
     <value>Cannot add item with key '{0}' to an existing '{1}'.</value>
   </data>
+  <data name="DirectiveExpectsIdentifierOrExpression" xml:space="preserve">
+    <value>The '{0}' directive expects an identifier or explicit razor expression.</value>
```

Comment:
Should we tell users here what "explicit razor expression" means so they know how to fix this error?